### ISSUE TITLE
Meaning of "reset" in "disconnect" and "terminate" events

### ISSUE THREAD

**Issue by int19h** [microsoft/debug-adapter-protocol#74]
The spec just says:
> A value of true indicates that this 'terminate' request is part of a restart sequence.

It is not clear what this means, exactly - the term _restart sequence_ is not defined anywhere. It seems like it is connected in some way to "restart" request, but it's not clear how they all interact.

(#73 is possibly related)

**Comment by haneefdm** [comment]
Yes, this is a bit unclear for me too. I am currently ignoring -- it never seems to be true -- probably because my extension supports a restart. I have a feeling that for extensions that don't support a restart request, a disconnect is called first before a full restart happens

**Comment by weinand** [comment]
"restart sequence" means:
terminate current debug session via the disconnect request and start a new session via the launch or attach request

**Comment by polinasok** [comment]
Wouldn't initialize (plus potentially more requests depending on how initialize event is implemented, see https://microsoft.github.io/debug-adapter-protocol/overview) be part of the restart sequence?

Would the restart sequence happen only if `supportsRestartRequest` is false? Otherwise, would restart just result in a single RestartRequest?
